src/Rings/MPolyQuo.jl

```diff
@@ -530,7 +524,7 @@ end
 
 
 function check_parent(a::MPolyQuoRingElem, b::MPolyQuoRingElem)
-  a.P == b.P || error("wrong parents")
+  @req parent(a) == parent(b) "parents must match"
```

Comment:
Hmmm actually perhaps this `@req` is not appropriate? But then again I am not sure because I don't know what the semantics of `check_parent` are supposed to be ... we have dozens and dozens of methods for it, but I couldn't find any documentation for it (@thofma @fieker ?)

Reply:
Why should the `@req` not be appropriate?

Reply:
OK, I just discovered that `check_parent` is described in the *Nemo* documentation, but not the AA documentation where I was searching (since AA uses `check_parent` extensively, I assumed it'd be explained there or nowhere).

(Funnily, that documentation suddenly uses the name `parent_check` instead, presumably that was an older name for it)

As to why `@req` might not be appropriate: well I wasn't sure what it is meant to be used for. But since it seems only input validation, I guess it'd be OK. Though, I should point that this `check_parent` methods used `error` before. And all other `check_parent` methods I checked (only a handful I randomly picked) also use `error`. So we are getting inconsistent. But OK, one step after the other.